lib/commons/dom/is-skip-link.js

```diff
@@ -32,14 +32,14 @@ dom.isSkipLink = function(element) {
 		axe._cache.set('firstPageLink', firstPageLink || null);
 	}
 
-	// if there are no page links then all all links will need to be
+	// if there are no page links then all offscreen links will need to be
 	// considered as skip links
 	if (!firstPageLink) {
-		return true;
+		return dom.isOffscreen(element);
```

Comment:
I'm not sure we could solve the two issues if the skip link is currently visible. Hidden skip link was the only solution that would identify them. Otherwise I think we have to move this rule to a best-practice and not fail it.